easybuild/framework/easyblock.py

```diff
@@ -358,34 +359,49 @@ def get_checksum_for(self, checksums, filename=None, index=None):
         :param filename: name of the file to obtain checksum for
         :param index: index of file in list
         """
-        checksum = None
-
-        # sometimes, filename are specified as a dict
+        chksum_input = filename
+        chksum_input_git = None
+        # if filename is provided as dict, take 'filename' key
         if isinstance(filename, dict):
-            filename = filename['filename']
+            chksum_input = filename['filename']
+            chksum_input_git = filename.get('git_config', None)
+        # early return if no filename given
+        if chksum_input is None:
+            self.log.debug("Cannot get checksum without a file name")
+            return None
+
+        if sys.version_info[0] >= 3 and sys.version_info[1] < 9:
+            # ignore any checksum for given filename due to changes in python/cpython#90021
```

Comment:
```suggestion
            # ignore any checksum for given filename due to changes in https://github.com/python/cpython/issues/90021

Reply:
fixed in https://github.com/easybuilders/easybuild-framework/pull/4660/commits/fb7535373fed06bda9f3375a11d6522f39eb7baa